src/client/proxy/http.cc

```diff
@@ -425,22 +407,41 @@ bool HTTPMessage::ExtractIncomingRequest() {
   return true;
 }
 
+void HTTPMessage::ParseURL()
+{
+  boost::network::uri::uri uri_parser("http://" + m_URL);
+  assert(uri_parser.is_valid());
```

Comment:
The new functions I write will throw on error instead of returning bool to signal success/failure. Thank you for the discussion of this design problem in IRC.